src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -241,10 +241,13 @@ ledgerFromRequest(T& ledger, JsonContext& context)
         }
         else
         {
+            int iVal;
             if (index.empty() || index == "current")
                 return getLedger(ledger, LedgerShortcut::CURRENT, context);
             else if (index == "closed")
                 return getLedger(ledger, LedgerShortcut::CLOSED, context);
+            else if (beast::lexicalCastChecked(iVal, indexValue.asString()))
```

Comment:
You don't need `indexValue.asString()` since you already have it: it's `index`.

Also, we should simplify the code a bit. There's a weird `if/else{if/else if/else}` hierarchy here that seems confusing. Please consider whether simplifying things helps make lines 238-255 more readable:

```
if (index.empty() || index == "current")
    return getLedger(ledger, LedgerShortcut::CURRENT, context);

if (index == "validated")
    return getLedger(ledger, LedgerShortcut::VALIDATED, context);

if (index == "closed")
    return getLedger(ledger, LedgerShortcut::CLOSED, context);

if (std::uint32_t x; beast::lexicalCastChecked(x, index))
    return getLedger(ledger, x, context);

return {rpcINVALID_PARAMS, "ledgerIndexMalformed"};
```

Note too that there's an `isNumeric` check on lines 231-233 could simply be eliminated and made to flow through to this code, simplifying the logic further (albeit at the cost of a little more work to convert the integer to a string and then back into an integer).